relation_engine/taxa/config.py

```diff
@@ -0,0 +1,134 @@
+"""
+A configuration parser for taxa delta loaders. The configuration is expected to be in TOML
+(https://toml.io/en/) format.
+"""
+
+import tomli  # TODO swap to stdlib in py 3.11
+from frozendict import frozendict
+from typing import Optional, BinaryIO
+from pathlib import Path
+
+# This could potentially be extended to the obograph loader as well, but would require more
+# inputs. Maybe extend? Might be better to duplicate
+
+# Since this is KBase internal code we can be a bit less compassionate re good
+# error messages, e.g. throwing any TOML parsing errors, etc.
+# As a result we get slightly less code to maintain and a completely trivial performance boost.
+# And there was much rejoicing.
+
+_SEC_ARANGO = "Arango"
+_SEC_VERSIONING = "Versioning"
+
+
+class DeltaLoaderConfig:
+    """
+    The delta load configuration parsed from a TOML configuration file. Once initialized, this
+    class will contain the fields:
+
+    inputs: dict[str, Path] - a dict with an entry for each input key provided in the
+        constructor.
+
+    url: str - the URL of an arango coordinator.
+    database: str - the name of the ArangoDB database to update.
+    user: str | None - the username, if any, of the user name to use when connecting to
+         ArangoDB.
+    password: str | None - the password for the user. Present IFF the user is present.
+    load_registry_collection: str - The name of the ArangoDB collection in which to register
+        the load.
+    node_collection: str - the name of the ArangoDB collection in which to load taxa nodes.
+    edge_collection: str - the name of the ArangoDB collection in which to load taxa edges.
+    merge_edge_collection: str | None - the name of the ArangoDB collection in which to
+        load merge edges.
+
+    load_version: str - the version of the load.
+    load_timestamp: int - the timestamp of the load in epoch milliseconds, e.g. when the load
+        will start to exist in the database.
+    release_timestamp: int - the timestamp, in unix epoch milliseconds, when the data was
+        released at the source.
+    """
+
+    def __init__(
+            self,
+            config_file: BinaryIO,
+            input_keys: list[str],
+            require_merge_collection: bool = False):
+        """
+        Create the configuration parser.
+
+        config_file - an open file-like object, opened in binary mode, containing the TOML
+            config file data.
+        input_keys - a list of keys for entries in the configuration that will
```

Comment:
if it was like ^ that one could default to all input keys, but allow specifying some subset. As of now I'm not sure one can enumerate the inputs in a config without knowledge of the names of the "input_keys" (I also worry that someone might name an input key something that later gets turned into a meaningful config key, but)

Reply:
```toml
[Inputs]
key1 = "path1"
key2 = "path2"
```
would also give the enumerability benefit i believe

Reply:
`input_keys` are the keys in the toml file where the values are the files or directories to process. See, for example, the top level keys in the two example toml files.

> Why is it an arg?

Because the keys aren't the same for NCBI and GTDB, and probably won't be the same for other loaders. Some loaders may require different numbers of files as well.

> I also worry that someone might name an input key something that later gets turned into a meaningful config key

They are meaningful config keys.

Does that answer your questions?

Reply:
If it doesn't maybe we should do a quick huddle, and either figure out how to change them or clarify the documentation

Reply:
Discussed by voice, agreed to put the input keys in their own [Inputs] section to avoid potential clobbering via future configuration parser changes